flutter-for-ios.md

```diff
@@ -27,3 +21,1847 @@ that are most relevant to your needs.
+you're doing I/O-bound work, such as a disk access or a network call, then
+you can safely just use `async`/`await` and you're all set. If, on the other
+hand, you need to do computationally intensive work that keeps the CPU busy,
+you want to move it to an `Isolate` as to avoid blocking the event loop.
+
+For I/O-bound work, you can declare the function as an `async` function
+and `await` on long-running tasks in the function:
+
+<!-- skip -->
+{% prettify dart %}
+loadData() async {
+  String dataURL = "https://jsonplaceholder.typicode.com/posts";
+  http.Response response = await http.get(dataURL);
+  setState(() {
+    widgets = json.decode(response.body);
+  });
+}
+{% endprettify %}
+
+This is how you would typically do network or database calls, which are both
+I/O operations.
+
+However, there are times where you may be processing a large amount of data and
+your UI could hang. In Flutter it is possible to take advantage of multiple CPU
+cores to do long-running or computationally intensive tasks. This is done by
+using `Isolate`s.
+
+Isolates are separate execution threads that run and do not share any memory
+with the main execution memory heap. This means you can’t access variables from
+the main thread or update your UI by calling `setState()`. Isolates are true to
+their name, and cannot share memory (e.g. in the form of static fields).
+
+Let's see an example of a simple isolate and how you can communicate and share
+data back to the main thread to update your UI.
+
+<!-- skip -->
+{% prettify dart %}
+loadData() async {
+  ReceivePort receivePort = new ReceivePort();
+  await Isolate.spawn(dataLoader, receivePort.sendPort);
+
+  // The 'echo' isolate sends its SendPort as the first message
+  SendPort sendPort = await receivePort.first;
+
+  List msg = await sendReceive(sendPort, "https://jsonplaceholder.typicode.com/posts");
+
+  setState(() {
+    widgets = msg;
+  });
+}
+
+// The entry point for the isolate
+static dataLoader(SendPort sendPort) async {
+  // Open the ReceivePort for incoming messages.
+  ReceivePort port = new ReceivePort();
+
+  // Notify any other isolates what port this isolate listens to.
+  sendPort.send(port.sendPort);
+
+  await for (var msg in port) {
+    String data = msg[0];
+    SendPort replyTo = msg[1];
+
+    String dataURL = data;
+    http.Response response = await http.get(dataURL);
+    // Lots of JSON to parse
+    replyTo.send(json.decode(response.body));
+  }
+}
+
+Future sendReceive(SendPort port, msg) {
+  ReceivePort response = new ReceivePort();
+  port.send([msg, response.sendPort]);
+  return response.first;
+}
+{% endprettify %}
+
+Here, `dataLoader()` is the `Isolate` that runs in its own separate execution thread.
+In this isolate you can do more CPU intensive processing, for example parsing a big
+JSON, or doing computationally intensive math such as encryption or signal processing.
+
+A full example that you can run is below.
+
+<!-- skip -->
```

Comment:
If it's a full example, don't use skip. We want the analyzer to check it. We only use that for partial code that will fail in the analyzer.

Reply:
We "took inspiration" (as in, lifted this) from the "Flutter for Android developers" page. Should we have it removed on Android too?